sktime/forecasting/compose/_by_group_forecaster.py

```diff
@@ -0,0 +1,526 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements a compositor to utilize forecasters based on ADI/CV categorization."""
+
+__author__ = ["shlok191"]
+
+from sktime.base._meta import _HeterogenousMetaEstimator
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.croston import Croston
+from sktime.forecasting.naive import NaiveForecaster
+from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.adi_cv import ADICVTransformer
+
+
+class ByGroupForecaster(BaseForecaster, _HeterogenousMetaEstimator):
```

Comment:
the name still feels a bit uninformtaive to me.

It should imply that the groups are computed from the time series. Not sure how to do that, that might be a good question to brainstorm.

Reply:
I agree. I wanted to have a name that captures the 2-stage nature of this with something that could showcase that there is a `series-to-primitives` transformer and then a forecaster but I just cannot think of a name that is not a mouthful.

Should I drop this question in dev-chat? Maybe someone has some good ideas

Reply:
How about something like `TransformSelectForecaster` or `PrimitivesDerivedForecaster`?

Reply:
Sounds good! I'm planning on pushing in a new commit with the added changes soon and with the name `TransformSelectorForecaster` and I'll request folks to pitch in ideas on Discord. If we get something new I'll add it in!

Also, yes. I'll add an example too when I write the message!